client/src/includes/initKeyboardShortcuts.ts

```diff
@@ -0,0 +1,38 @@
+import Mousetrap from 'mousetrap';
+
+/**
+ * Set up the keyboard shortcuts based on any element that has `data-keyboard-shortcut`.
+ * Ensure that any duplicates are ignored (the first element in the DOM will get the shortcut).
+ * Ensure that we focus on the element 'triggered' (button or input) and if it is a button
+ * trigger the button's click method.
+ *
+ * Finally, return the resolved shortcuts object.
+ */
+const initKeyboardShortcuts = () => {
+  const elements = document.querySelectorAll('[data-keyboard-shortcut]');
+
+  const keyboardShortcuts = [...elements]
+    .filter((element): element is HTMLElement => element instanceof HTMLElement)
+    .map((element): [string | null, HTMLElement] => {
+      const { keyboardShortcut = null } = element.dataset;
+      return [keyboardShortcut, element];
+    })
+    .filter(([keyboardShortcut]) => keyboardShortcut)
+    .reduce((acc, [keyboardShortcut, element]) => {
+      if (!keyboardShortcut || acc[keyboardShortcut]) return acc;
+
+      /** add keyboard shortcut */
+      Mousetrap.bind([keyboardShortcut], (event) => {
+        event.preventDefault();
+        element?.focus();
+        if (element.tagName === 'BUTTON') element?.click();
```

Comment:
I don’t understand when this would happen without an element being defined. Hopefully with the addition of `<HTMLElement>` to the qSA this should no longer be needed.

```suggestion
        element.focus();
        if (element.tagName === 'BUTTON') element.click();
```

Reply:
thanks @thibaudcolas  I went with a different tact and have an update now.